src/foundry/common/utils/bitmask.d.mts

```diff
@@ -0,0 +1,147 @@
+export default class BitMask extends Number {
+  /**
+   * Create a new BitMask instance.
+   * @param states    - An object containing valid states and their corresponding initial boolean values (default is null).
+   */
+  constructor(states?: Record<string, boolean> | null);
```

Comment:
Can this type be more specific? I can check Foundry's source myself if you want but I'd be inclined to say there's probably only a few valid keys here.

In fact everywhere there's a `string` representing a state I'd bet it can actually represent something more specific.

Reply:
From what I can tell it (inside this class at least) will take any arbitrary string.  The only constraint is that states can't have more than 32 keys, though I have no idea how to type that constraint.  The class appears to let you basically create a bitmask stored as a 32-bit number for up to 32 boolean states that you want to track. I assume this is to then allow those states to be saved somewhere more efficienctly (because creating a class to handle a single 32-bit value seems like a lot of overhead :))

Reply:
You could _technically_ type that but it'd probably just reach a type complexity limit anyways. Sounds like for all intents and purposes it's arbitrary. Thanks!